src/asm/asmy.y

```diff
@@ -101,7 +101,11 @@ static uint32_t str2int2(char *s, int32_t length)
 	int32_t i;
 	uint32_t r = 0;
 
-	i = length < 4 ? 0 : length - 4;
+	i = 0;
+	if (length > 4) {
+		warning(WARNING_LONG_STR, "String is too long in expression '%s' only last 4 characters will be used.", s)
```

Comment:
`"String '%s' is too long, conversion will use only last 4 characters"`

Reply:
Actually, I think mentioning "charmap conversion" explicitly would perhaps help users understand why they're getting this warning. OTOH they may not have read TFM, so they may still have no idea what this is, and suggesting `(Did you mean to use "STRCMP"?)` may be more explicit? What do you think?

Reply:
If you want to go for the best way to help newbies, I'd say split this into two warnings:

- One warning whenever multi-codepoint constants are used in non-string contexts. This is basically GCC's `-Wmultichar`, a warning for something that is so likely to be invalid that it is enabled by default. RGBDS might want to enable it under `-Wall` just in case someone has been using these constants all along.
- Another warning, definitely enabled by default, for the situation when the string is actually truncated. This isn't really four _characters_, but rather four _codepoints_, so I'd rather avoid mentioning characters at all — `warning: string "%s" truncated to "%s" in expression` should do the trick.

Reply:
Problem with the first warning you suggest is that:
`cp "<END_OF_TEXT>"`
Is valid code, and I think the reason why string can be converted to integer expressions like this.

Reply:
Also just realized that it will print the result of the charmap conversion right now, not the input...

Reply:
That's why I said "multi-codepoint". That string is (probably) one codepoint in the charmap. (And if it's not, the warning is definitely necessary there.)

Reply:
I agree that warning if a string containing more than one *codepoint* gets converted, a warning should be raised. The order the bytes are output in is not exactly clear, either—I think it's little-endian, such that the following two are equivalent:
```
db "WHEE"
``````
YAHOO equ "WHEE"
db YAHOO
```

That said, in the worst case, these two don't match. What should be we do, then?

Reply:
Wouldn't that last case truncate to _one_ byte?

Reply:
The last one gives a warning:
https://daid.github.io/rgbds-live/#FAZQogwgKgkg8gOQAQCIBGBDAdgawAx4oA0SASnALJ4DaAJAQQLrACaAggBJxxICmAjgFdUAdQ5gwKYEiQATNEnZc4QA

Reply:
Indeed.

Reply:
Oh and:
```
YAHOO equ "WHEE"
db YAHOO
```
Equals to:
```
db "E"
```
Not "W" as you would expect.